source/Halibut/Transport/Protocol/ITypeRegistry.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System;
+
+namespace Halibut.Transport.Protocol
+{
+    public interface ITypeRegistry
+    {
+        void Register(params Type[] registeredServiceTypes);
+        void RegisterType(Type type, string path, bool ignoreObject);
+        bool IsInAllowedTypes(Type type);
```

Comment:
The name `IsInAllowedTypes` is a little confusing, because without looking into it, the reader wonders what an 'allowed type' is.

Some suggestions:
 - Contains (keeping with standard .NET method names)
 - CanSerialize
 - IsRegistered (although this makes more sense if we rename the interface)